weave-js/src/common/components/WBMenu.tsx

```diff
@@ -106,130 +100,69 @@ export const WBMenu = React.forwardRef<HTMLDivElement, WBMenuProps>(
+      if (!highlightFirst) return [undefined, undefined];
+      const foundIndex = options.findIndex(el => !el.disabled);
 
-      return found >= 0 ? found : 0;
-    }, [highlightFirst, options, selected]);
-    // dedupe options
-    const prevLength = options.length;
-    options = _.uniqBy(options, opt => opt.value);
-    if (options.length !== prevLength) {
-      // eslint-disable-next-line no-console
-      console.warn('Passed duplicate options into WBMenu');
-    }
-
-    const [highlighted, setHighlighted] = useControllableState(
-      options[getDefaultHoverIndex()]?.value || null,
-      highlightedProp,
-      onChangeHighlighted
-    );
-    const highlightedIndex = useMemo(() => {
-      if (highlighted === null) {
-        return null;
-      }
-      const found = options.findIndex(opt => highlighted === opt.value);
+      if (!foundIndex) return [undefined, undefined];
 
-      return found >= 0 ? found : null;
-    }, [options, highlighted]);
+      return [options[foundIndex]?.value, foundIndex];
+    }, [highlightFirst, options]);
 
-    const lastHighlightedValue = React.useRef<string | number | null>(
-      highlighted
-    );
-    const setHighlightedWrapped = React.useCallback(
-      (newHighlighted: number | string | null) => {
-        const last = highlighted;
-        setHighlighted(newHighlighted);
-        lastHighlightedValue.current = last;
-      },
-      [highlighted, setHighlighted]
+    const [highlighted, setHighlighted] = useState<string | number | undefined>(
+      defaultHighlighted
     );
+    const [highlightedIndex, setHighlightedIndex] = useState<
+      number | undefined
+    >(defaultHighlightedIndex);
 
     const contentRef = React.useRef<HTMLDivElement>(null);
-    React.useEffect(() => {
+    useEffect(() => {
       function onKeyDown(e: KeyboardEvent) {
-        if (e.keyCode === 38 /* up */) {
+        if (['ArrowUp', 'ArrowDown'].includes(e.key)) {
           e.preventDefault();
-          let moved = false;
-          for (
-            let i = (highlightedIndex || getDefaultHoverIndex()) - 1;
-            i >= -1;
-            i--
-          ) {
-            const modIndex =
-              ((i % options.length) + options.length) % options.length;
-            if (!options[modIndex].disabled) {
-              setHighlightedWrapped(options[modIndex].value);
-              const child = contentRef.current?.children[modIndex];
-              if (child) {
-                scrollIntoView(child, {
-                  scrollMode: 'if-needed',
-                  block: 'start',
-                });
-              }
-              moved = true;
-              break;
+          if (!options || !options.length) return;
+
+          const block = e.key === 'ArrowUp' ? 'start' : 'end';
+          const moveAmount = e.key === 'ArrowUp' ? -1 : 1;
+
+          let newHighlightedIndex;
+          if (typeof highlightedIndex === 'undefined') {
```

Comment:
why can't we access the actual value here? ie: `highlightedIndex == null`

Reply:
or just initialize it with type: `string | number | undefined`. or maybe even make that a type, which can be referenced in multiple places.

Reply:
I pretty much never use type coercion unless it is the shorthanded `!someVar`, but in this case I can't do that. Also, I find the explicit approach more legible and less prone to bugs. As for the type aliasing, I typically don't use them unless it is a super complex thing but happy to add it if the weave team prefers them.

I could check for `hightlightedIndex === undefined` if you'd prefer to not use `typeof`

Reply:
Going to use >= 0 here to account for unfound index